### PR TITLE
Setting memory limits via snakemake "resource" keyword.

### PR DESCRIPTION
**Pull request by SilasK**
I finished this. I set the memory limit via the snakemake `resource` keyword, instead of params.

I modified the all snakefiles and the config file accordingly.

### CODE REVIEW

File: atlas/rules/assemble.snakefile [hunk 1]
```diff
@@ -255,7 +265,8 @@ if config.get("assembler", "megahit") == "megahit":
                    --min-count {params.min_count} \
                    --merge-level {params.merge_level} \
                    --prune-level {params.prune_level} \
-                   --low-local-ratio {params.low_local_ratio}"""
+                   --low-local-ratio {params.low_local_ratio} \
+                   --memory {resources.mem}000000000 #GB"""
```
Comment (brwnj) [inline atlas/rules/assemble.snakefile]: Need to move the GB comment outside of the string.

**Comment by SilasK** [comment]
requested chages are made, ready for merg. @brwnj

**Comment by SilasK** [comment]
I will probably merge this after the paired-end workflow is finished #27

**Comment by brwnj** [comment]
Merge this as needed so you can get it tested along with the other major changes.

**Pull request merged**